pkg/alert/commands.go

```diff
@@ -318,3 +316,187 @@ func NewDeleteAlertSettingConditionCommand() *cobra.Command {
+
+	return cmd
+}
+
+//NewDeleteAlertChannelCommand creates `alertchannels delete` command
+func NewDeleteAlertChannelCommand() *cobra.Command {
+	var (
+		channelID string
+	)
+
+	cmd := &cobra.Command{
+		Use:              "delete",
+		Short:            "Delete an alert channel",
+		Example:          `alertchannels delete --channelID="fa0e9b96-1048-4f4c-b776-4e96ca62f37d"`,
+		TraverseChildren: true,
+		SilenceErrors:    true,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			err := config.Client.DeleteAlertChannel(channelID)
+			if err != nil {
+				golog.Errorf("Failed to delete alert channel [%s]. [%s]", channelID, err.Error())
+				return err
+			}
+			return bite.PrintInfo(cmd, "Alert channel [%s] deleted", channelID)
+		},
+	}
+
+	cmd.Flags().StringVar(&channelID, "channelID", "", "Channel ID")
+	cmd.MarkFlagRequired("channelID")
+	bite.CanBeSilent(cmd)
+
+	return cmd
+}
+
+// NewCreateAlertChannelCommand creates `alertchannels create` command
+func NewCreateAlertChannelCommand() *cobra.Command {
+	var (
+		propertiesRaw string
+		channel       = api.AlertChannelPayload{}
+	)
+
+	cmd := &cobra.Command{
+		Use:              "create",
+		Short:            "Create a new alert channel",
+		Example:          `alertchannels create --name="kafka-prd-health" --templateName="Slack" --connectionName="slack-connection" --properties="[{\"key\":\"username\",\"value\":\"@luk\"},{\"key\":\"channel\",\"value\":\"#lenses\"}]"`,
+		SilenceErrors:    true,
+		TraverseChildren: true,
+		RunE: func(cmd *cobra.Command, args []string) error {
+
+			if err := bite.CheckRequiredFlags(cmd, bite.FlagPair{"name": channel.Name, "connectionName": channel.ConnectionName, "templateName": channel.TemplateName, "properties": channel.Properties}); err != nil {
+				return err
+			}
+
+			if propertiesRaw != "" {
+				if err := bite.TryReadFile(propertiesRaw, &channel.Properties); err != nil {
+					// from flag as json.
+					if err = json.Unmarshal([]byte(propertiesRaw), &channel.Properties); err != nil {
+						return fmt.Errorf("Unable to unmarshal the properties: [%v]", err)
+					}
+				}
+			}
+
+			if err := config.Client.CreateAlertChannel(channel.Name, channel.ConnectionName, channel.TemplateName, channel.Properties); err != nil {
+				golog.Errorf("Failed to create alert channel [%s]. [%s]", channel.Name, err.Error())
+				return err
+			}
+
+			return bite.PrintInfo(cmd, "Alert channel [%s] created", channel.Name)
+		},
+	}
+
+	cmd.Flags().StringVar(&channel.Name, "name", "", "Alert channel name")
+	cmd.Flags().StringVar(&channel.ConnectionName, "connectionName", "", "Alert channel connection name")
+	cmd.Flags().StringVar(&channel.TemplateName, "templateName", "", "Alert channel template name")
+	cmd.Flags().StringVar(&propertiesRaw, "properties", "", `Alert channel properties .e.g. "[{\"key\":\"username\",\"value\":\"@luk\"},{\"key\":\"channel\",\"value\":\"#lenses\"}]"`)
+	bite.CanBeSilent(cmd)
+	bite.Prepend(cmd, bite.FileBind(&channel))
```

Comment:
How is this used?

Reply:
Necessary for the `alertchannels create <yaml file>` .